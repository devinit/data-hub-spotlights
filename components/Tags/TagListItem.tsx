import React, { FunctionComponent } from 'react';
import { Button } from '../Button';

interface TagsListItemProps {
  label: string;
  onRemove?: (label: string) => void;
}

const TagListItem: FunctionComponent<TagsListItemProps> = ({ label, onRemove: onRemoveTag }) => {
  const onRemove = (): void => {
    if (onRemoveTag) {
      onRemoveTag(label);
    }
  };

  return (
    <li className="m-pills__item">
      {label.toLowerCase()}
      <Button onClick={onRemove} className="">
        <i role="presentation" aria-hidden="true" className="ico ico--20 ico-cross-slate"></i>
      </Button>
      <style jsx>{`
        text-transform: capitalize;
      `}</style>
    </li>
  );
};

export { TagListItem };
