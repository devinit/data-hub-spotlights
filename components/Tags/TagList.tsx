import React, { FunctionComponent, Children, isValidElement } from 'react';
import { TagListItem } from './TagListItem';

export const TagList: FunctionComponent = ({ children }) => (
  <ul className="m-pills">
    {Children.map(children, (child) => (isValidElement(child) && child.type === TagListItem ? child : null))}
  </ul>
);
