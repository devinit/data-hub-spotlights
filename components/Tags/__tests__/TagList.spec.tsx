/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { TagList } from '../TagList';
import { TagListItem } from '../TagListItem';

describe('TagList', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(<TagList />).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders TagListItems', () => {
    const renderer = TestRenderer.create(
      <TagList>
        <TagListItem label="Tag 1" />
        <div>Not TagListItem</div>
        <TagListItem label="Tag 2" />
      </TagList>
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
