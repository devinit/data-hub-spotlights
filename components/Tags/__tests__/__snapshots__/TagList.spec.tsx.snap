// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagList renders TagListItems 1`] = `
<ul
  className="m-pills"
>
  <li
    className="m-pills__item"
  >
    tag 1
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      <i
        aria-hidden="true"
        className="ico ico--20 ico-cross-slate"
        role="presentation"
      />
    </button>
    <style
      jsx={true}
    >
      
        text-transform: capitalize;
      
    </style>
  </li>
  <li
    className="m-pills__item"
  >
    tag 2
    <button
      className=""
      onClick={[Function]}
      type="button"
    >
      <i
        aria-hidden="true"
        className="ico ico--20 ico-cross-slate"
        role="presentation"
      />
    </button>
    <style
      jsx={true}
    >
      
        text-transform: capitalize;
      
    </style>
  </li>
</ul>
`;

exports[`TagList renders correctly 1`] = `
<ul
  className="m-pills"
/>
`;
